src/main/java/seedu/address/model/person/TelegramAddress.java

```diff
@@ -10,11 +10,11 @@
 public class TelegramAddress {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Names should only contain alphanumeric characters, numbers and underscores, "
+            "Telegram addresses should only contain alphanumeric characters, numbers and underscores, "
                     + "and it should not be blank";
 
 
-    public static final String VALIDATION_REGEX = "^\\S[A-Za-z0-9_]*$";
+    public static final String VALIDATION_REGEX = "^[A-Za-z0-9]+[\\w]{3,}[A-Za-z0-9]+$";
```

Comment:
```suggestion
    public static final String VALIDATION_REGEX = "^[A-Za-z0-9][\\w]{3,}[A-Za-z0-9]$";
```
For aesthetic. No real impact.